### PR TITLE
Add infrastructure for running the librarian inside of Docker

### PR DESCRIPTION
**Pull request by plaplant**
This PR adds the ability to run the Librarian as a series of docker containers. It treats the librarian server as three separate subsystems:
* the frontend app running the tornado webserver
* the backing PostgreSQL database
* a librarian store that hosts data and is running an `openssh-server` instance

The app and store images are built off of the `miniconda3` docker image, and use `conda` to manage the python environment. The Postgres image uses an off-the-shelf image and requires no additional modification. (Annoyingly, an off-the-shelf solution for the store/ssh server could not be used because the `hera_librarian` package is required for getting data about the saved files.) Some additional configuration is required of the user (primarily around generating ssh keys and configurations), which is briefly outlined in a new section of the docs.

Local testing confirms that accessing the webapp inside of the running container works as expected, as well as uploading and searching. The SQL database and uploaded files persist between docker instances, so the implementation seems stable. Further configuration by the user will almost certainly be required, but hopefully the containers as defined provide a sufficient starting point.

Now that this containerized infrastructure exists, it opens the possibility of designing and running more sophisticated unit/integration tests that exercise more of the functionality of the librarian. However, these improvements are left to future work.

### CODE REVIEW

**Comment by pkgw** [comment]
Cool. Is there anything here that you think I ought to review closely?

**Comment by plaplant** [comment]
@pkgw sorry, missed your comment initially. If you don't mind reading the updated docs to see if they make sense, that would be excellent. I'd also be interested to hear if you have suggestions or comments on how the ssh key handling is done, as I mostly made it up and don't know if it's optimal.

File: container/wait-for-it.sh [hunk 1]
```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bash
+# Use this script to test if a given TCP host/port are available
```
Comment (pkgw) [inline container/wait-for-it.sh:2]: Is this file copied from somewhere? If so, it wouldn't hurt to link to the original source in a comment for both credit and knowing where to find updates in the future.

**Review by pkgw — APPROVED** [review]
I think this all looks good.

I think what you're doing with SSH is definitely reasonable. I suppose one alternative might be to just build in [Vagrant's insecure passwordless SSH keys](https://github.com/hashicorp/vagrant/tree/master/keys) instead of having the user generate a key themselves. This opens you up to abuses if someone on your system knows that you're running these containers, though, and it seems better to err on the side of being a bit more secure.

Doc updates look good.

File: container/wait-for-it.sh (hunk 1 above)
Comment (plaplant) [reply container/wait-for-it.sh:2]: Yes, I did ~steal~ copy it from somewhere. I'll add a comment explaining its provenance.

**Comment by plaplant** [comment]
@pkgw thanks very much for the review! I've added an attribution for the script. I'm going to go ahead and merge this.

**Pull request merged**